frontend/src/pages/petlistings/PetDetail.jsx

```diff
@@ -1,8 +1,244 @@
+      headers: {
+        Authorization: `Bearer ${token}`,
+        'Content-Type': "application/json",
+      }
+    }
+
+    const fetchAllData = async () => {
+      try {
+        const petlisting = await getPetlisting();
+        setListingData(petlisting);
+
+        const listingOwner = await getListingOwner(petlisting.owner);
+        setShelterData(listingOwner);
+
+      } catch (err) {
+        setErr(err.message);
+        console.log("Uh oh =>", err);
+      }
+
+      setLoading(false);
+    }
+
+    const getPetlisting = async () => {
+      const res = await fetch(listingUrl, getReqContent);
+
+      if (res.status !== 200) {
+        throw new Error(res.statusText);
+      }
+      const listing = await res.json();
+      return listing.data;
+    }
+
+    const getListingOwner = async (email) => {
+      const allShelters = [];
+      let hasNext = true;
+      let page = 1;
+      while (hasNext) {
+        let shelterUrl = `${process.env.REACT_APP_API_URL}/api/shelters?page=${page}`;
+        const res = await fetch(shelterUrl, getReqContent);
+
+        if (res.status != 200) {
+          throw new Error(res.statusText);
+        }
+
+        const shelters = await res.json();
+        allShelters.push(...shelters.data);
+        hasNext = shelters.page.has_next;
+        page++;
+      }
+      const ownerShelter = allShelters.find(s => s.email === email);
+      return ownerShelter;
+    }
+
+    
+
+    fetchAllData();
+    // end of useEffect()
+  }, []);
+
+
+  const processPhotos = (photos) => {
+    const defaultPhoto = "/images/logo_ref.png";
+    if (photos.length == 0) {
+      return [defaultPhoto];
+    }
+
+    const photoUrls = photos.map((photo) => {
+      return `${process.env.REACT_APP_API_URL}/${photo.url}`;
+    });
+    return photoUrls;
+  }
+
+
+
+  const mainContent = (listing, shelter) => {
+    const fDate = new Date(listing.created_time).toLocaleDateString();
+    const allPhotos = processPhotos(listing.photos);
+    const isListingOwner = userId == shelter.id;
+
+    const renderActionBtn = () => {
+      if (isListingOwner) {
+        // return edit button
+        return (
+          <Button handleClick={() => {navigate(`/petlistings/${petId}/edit`)}}>Edit pet</Button>
+        )
+      }
+      
+      const shelterContact = `/messages/${shelter.id}`
+      if (listing.status !== "AV" ) {
+        return (
+          // contact shelter btn
+          <>
+             <Button classes="square-btn" id="contact-shelter-btn" handleClick={() => {navigate(shelterContact)}}>contact shelter</Button>
+          </>
+        )
+      }
+      
+      return (
+        // returning:
+        // contact shelter btn
+        // application form btn
+        <>
+        <Button classes="square-btn" id="contact-shelter-btn" handleClick={() => {navigate(shelterContact)}}>contact shelter</Button>
+        <Button id="application-form-btn" handleClick={() => {navigate(`/applications`)}}>Application Form</Button>
```

Comment:
<img width="539" alt="Screen Shot 2023-12-04 at 6 51 28 PM" src="https://github.com/lujorryn/PetPal-G0346/assets/144406716/8d6abe21-c856-4d5f-bb46-44bfed0b4779">

For shelter users these 2 buttons should not appear on other shelters' petlistings